ci/scripts/cpp_test.sh

```diff
@@ -89,14 +89,21 @@ pushd ${build_dir}
 if [ -z "${PYTHON}" ] && ! which python > /dev/null 2>&1; then
   export PYTHON="${PYTHON:-python3}"
 fi
-ctest \
+if [ "${ARROW_USE_MESON:-OFF}" = "ON" ]; then
+  ARROW_BUILD_EXAMPLES=OFF # TODO: Remove this
+  meson test \
+    --print-errorlogs \
+    "$@"
```

Comment:
We aren't building any tests now, are we? On the current CI:
```
+ meson test --print-errorlogs
No tests defined.
```

Reply:
For now we aren't actually building any tests - just creating an MRE for Meson with libarrow

Reply:
Right. We'll add tests later.